python/python/tests/test_json.py

```diff
@@ -0,0 +1,348 @@
+        )
+
+        dataset = lance.write_dataset(overwrite_table, dataset_path, mode="overwrite")
+
+        # Verify overwrite
+        result = dataset.to_table()
+        assert result.num_rows == 1
+        assert result.column("id").to_pylist() == [3]
+
+
+def test_json_filter_and_select():
+    """Test filtering and selecting with JSON columns."""
+
+    json_data = [
+        {"user_id": 1, "preferences": {"theme": "dark"}},
+        {"user_id": 2, "preferences": {"theme": "light"}},
+        {"user_id": 3, "preferences": {"theme": "dark"}},
+    ]
+
+    json_strings = [json.dumps(d) for d in json_data]
+    json_ext_array = JsonArray.from_json_strings(json_strings)
+
+    table = pa.table(
+        {
+            "id": pa.array([1, 2, 3]),
+            "user_data": json_ext_array,
+            "active": pa.array([True, False, True]),
+        },
+        schema=pa.schema(
+            [
+                pa.field("id", pa.int32()),
+                json_field("user_data"),
+                pa.field("active", pa.bool_()),
+            ]
+        ),
+    )
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        dataset_path = Path(tmpdir) / "filter_test.lance"
+        lance.write_dataset(table, dataset_path, data_storage_version="2.2")
+        dataset = lance.dataset(dataset_path)
+
+        # Test column selection
+        selected = dataset.to_table(columns=["id", "user_data"])
+        assert selected.num_columns == 2
+        assert "active" not in selected.column_names
+
+        # Test filtering
+        filtered = dataset.to_table(filter="active = true")
+        assert filtered.num_rows == 2
+        assert filtered.column("id").to_pylist() == [1, 3]
+
+
+def test_json_null_handling():
+    """Test handling of null JSON values."""
+
+    # Mix of valid JSON and nulls
+    json_strings = [
+        json.dumps({"key": "value1"}),
+        None,
+        json.dumps({"key": "value2"}),
+        None,
+        json.dumps({"key": "value3"}),
+    ]
+
+    json_ext_array = JsonArray.from_json_strings(json_strings)
+
+    table = pa.table(
+        {"id": pa.array(range(5)), "optional_data": json_ext_array},
+        schema=pa.schema(
+            [pa.field("id", pa.int32()), json_field("optional_data", nullable=True)]
+        ),
+    )
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        dataset_path = Path(tmpdir) / "null_test.lance"
+        lance.write_dataset(table, dataset_path, data_storage_version="2.2")
+        dataset = lance.dataset(dataset_path)
+
+        result = dataset.to_table()
+        assert result.num_rows == 5
+
+        # Check null mask
+        data_column = result.column("optional_data")
+        assert data_column.null_count == 2
+        assert data_column.is_null().to_pylist() == [False, True, False, True, False]
+
+
+def test_json_large_documents():
+    """Test handling of large JSON documents."""
+
+    # Create a moderately large JSON document
+    large_doc = {
```

Comment:
Unless we are crossing the 2GB threshold or benchmarking I'm not sure this size is particularly interesting.

Reply:
Correct, let me just remove it.